### PR TITLE
[Android] Set the right API level in AndroidUtils

### PR DESCRIPTION
**Pull request by joseluismarti**
## Description
Display.Mode was added en API level 23, change 24 to 23.

Add some minor changes: clean unnecessary includes, set the type on some auto variables and small style changes.

## How has this been tested?
Tested on Android TV 9

## Types of change
- [X] **Bug fix** (non-breaking change which fixes an issue)
- [X] **Clean up** (non-breaking change which removes non-working, unmaintained functionality)
- [ ] **Improvement** (non-breaking change which improves existing functionality)
- [ ] **New feature** (non-breaking change which adds functionality)
- [ ] **Breaking change** (fix or feature that will cause existing functionality to change)
- [ ] **Cosmetic change** (non-breaking change that doesn't touch code)
- [ ] **None of the above** (please explain below)

## Checklist:
- [X] My code follows the **[Code Guidelines](https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md)** of this project 
- [ ] My change requires a change to the documentation, either Doxygen or wiki
- [ ] I have updated the documentation accordingly
- [X] I have read the **[Contributing](https://github.com/xbmc/xbmc/blob/master/docs/CONTRIBUTING.md)** document
- [ ] I have added tests to cover my change
- [ ] All new and existing tests passed

### CODE REVIEW

File: xbmc/windowing/android/AndroidUtils.cpp [hunk 1]
```diff
@@ -91,17 +84,17 @@ static void fetchDisplayModes()
...
-        s_res_cur_displayMode.fPixelRatio   = 1.0f;
+        s_res_cur_displayMode.fPixelRatio = 1.0f;
         s_res_cur_displayMode.strMode = StringUtils::Format(
             "{}x{} @ {:.6f}{} - Full Screen", s_res_cur_displayMode.iScreenWidth,
             s_res_cur_displayMode.iScreenHeight, s_res_cur_displayMode.fRefreshRate,
             s_res_cur_displayMode.dwFlags & D3DPRESENTFLAG_INTERLACED ? "i" : "");
 
         std::vector<CJNIDisplayMode> modes = display.getSupportedModes();
-        for (auto m : modes)
+        for (CJNIDisplayMode& m : modes)
```
Comment (enen92) [inline xbmc/windowing/android/AndroidUtils.cpp:97]:
```suggestion
        for (const CJNIDisplayMode& m : modes)
```
?

File: xbmc/windowing/android/AndroidUtils.cpp [hunk 2]
```diff
@@ -284,7 +277,7 @@ bool CAndroidUtils::ProbeResolutions(std::vector<RESOLUTION_INFO>& resolutions)
 
   if (s_hasModeApi)
   {
-    for(RESOLUTION_INFO res : s_res_displayModes)
+    for (RESOLUTION_INFO res : s_res_displayModes)
```
Comment (enen92) [inline xbmc/windowing/android/AndroidUtils.cpp:280]: is the copy intended?

Comment (joseluismarti) [reply xbmc/windowing/android/AndroidUtils.cpp:280]: better by reference, I change it, thanks

File: xbmc/windowing/android/AndroidUtils.cpp (hunk 1 above)
Comment (joseluismarti) [reply xbmc/windowing/android/AndroidUtils.cpp:97]: No, since the CJNIDisplayMode functions used below are not marked const

Comment (enen92) [reply xbmc/windowing/android/AndroidUtils.cpp:97]: hum to me it seems they could be marked const as they are simple getters

**Review by ksooo — APPROVED** [review]
LGTM

**Review by ksooo — CHANGES_REQUESTED** [review]
Please double-check that `for (RESOLUTION_INFO& res : s_res_displayModes)` - change from copy to reference -- actually works. If not sure, stay with the copy.

**Comment by joseluismarti** [comment]
I've checked that after the for, the ```s_res_displayModes``` vector is modified. Change to copy, thanks @ksooo

File: xbmc/windowing/android/AndroidUtils.cpp (hunk 2 above)
Comment (joseluismarti) [reply xbmc/windowing/android/AndroidUtils.cpp:280]: Change to copy as original, thanks

**Review by ksooo — APPROVED** [review]

**Comment by fuzzard** [comment]
Jenkins build this please

**Comment by fuzzard** [comment]
Windows failure unrelated, no need to rebuild.

**Pull request merged**